.gitignore

```diff
@@ -78,3 +78,8 @@ embedded_zookeeper/
 conf/log4j2.xml
 conf/kyuubi-defaults.conf
 conf/kyuubi-env.sh
+
+# For Antlr
+kyuubi-server/gen
```

Comment:
what is this for?

Reply:
Base lexer files generate useless files after compiled, ignore these files

Reply:
I tried to compile the `kyuubi-server` module locally based on this PR, and all the source files generated by ANTLR are under the folder `kyuubi-server/target/generated-sources/antlr4`

you can generate only antlr-related source files by running

```bash
./build/mvn antlr4:antlr4 -pl kyuubi-server
```

or compile the module `kyuubi-server` by running

```bash
./build/mvn clean package -DskipTests -Pspark-provided,flink-provided,hive-provided -am -pl kyuubi-server
```

Reply:
It looks like these files will be generated, this seems to be consistent with [Spark's behavior](https://github.com/apache/spark/blob/b5a9e1fee374b5d1e86021c7b49f18c5bb1b986e/.gitignore#L119-L122)
```
kyuubi-server % ./build/mvn clean package -DskipTests -Pspark-provided,flink-provided,hive-provided -am -pl kyuubi-server

kyuubi-server % pwd
/Users/yikaifei/apache/code/kyuubi/kyuubi-server

kyuubi-server % ll src/main/antlr4/org/apache/kyuubi/sql/
total 24
-rw-r--r--  1 yikaifei  staff  2830 12  8 16:35 KyuubiSqlBaseLexer.g4
-rw-r--r--  1 yikaifei  staff   297 12  8 16:37 KyuubiSqlBaseLexer.tokens
-rw-r--r--  1 yikaifei  staff  1231 12  8 16:37 KyuubiSqlBaseParser.g4
drwxr-xr-x  5 yikaifei  staff   160 12  8 16:12 gen

kyuubi-server % ll gen
total 48
-rw-r--r--  1 yikaifei  staff   7169 12  8 16:35 KyuubiSqlBaseLexer.interp
-rw-r--r--  1 yikaifei  staff  10030 12  8 16:35 KyuubiSqlBaseLexer.java
-rw-r--r--  1 yikaifei  staff    297 12  8 16:35 KyuubiSqlBaseLexer.tokens
```